ethereum-consensus/src/deneb/polynomial_commitments.rs

```diff
@@ -0,0 +1,152 @@
+use crate::{deneb::blob_sidecar::Blob, primitives::Bytes32, ssz::prelude::*};
+use c_kzg::Error;
+pub use c_kzg::KzgSettings;
+
+use super::mainnet::FIELD_ELEMENTS_PER_BLOB;
+
+pub const BYTES_PER_FIELD_ELEMENT: usize = 32;
+pub const BYTES_PER_BLOB: usize = BYTES_PER_FIELD_ELEMENT * FIELD_ELEMENTS_PER_BLOB;
+pub const BYTES_PER_COMMITMENT: usize = 48;
+pub const BYTES_PER_PROOF: usize = 48;
+
+pub type Bytes48 = ByteVector<48>;
+pub type VersionedHash = Bytes32;
+pub type FieldElement = Bytes32;
+pub type KzgCommitment = Bytes48;
+pub type KzgProof = Bytes48;
+
+pub struct ProofAndEvaluation {
+    pub proof: KzgProof,
+    pub evaluation: Bytes32,
+}
+
+pub fn blob_to_kzg_commitment<const BYTES_PER_BLOB: usize>(
+    blob: &Blob<BYTES_PER_BLOB>,
+    kzg_settings: &KzgSettings,
+) -> Result<KzgCommitment, Error> {
+    let blob = c_kzg::Blob::from_bytes(blob.as_ref()).unwrap();
+
+    let commitment = c_kzg::KzgCommitment::blob_to_kzg_commitment(&blob, kzg_settings)?;
+    let inner = ByteVector::try_from(commitment.to_bytes().as_slice()).unwrap();
```

Comment:
this use of `unwrap` should be fine but I'd consider `expect("correct size")` or something to indicate that we in fact expect this to have no error